app/services/course/duplication/object_duplication_service.rb

```diff
@@ -40,17 +40,48 @@ def time_shift(current_course, target_course)
   # @return [Object] The duplicated object, if `objects` is a single object.
   # @return [Array] Array of duplicated objects, if `objects` is an array.
   def duplicate_objects(objects)
-    # TODO: Inform the user when the duplication is complete.
+    # TODO: Email the user when the duplication is complete.
     Course.transaction do
       duplicated = duplicator.duplicate(objects)
-      success = duplicated.respond_to?(:save) ? duplicated.save : duplicated.all?(&:save)
-      raise ActiveRecord::Rollback unless success
+      save_success = duplicated.respond_to?(:save) ? duplicated.save : duplicated.all?(&:save)
+      after_save_success = save_success && after_save(objects, duplicated)
+      raise ActiveRecord::Rollback unless after_save_success
       duplicated
     end
   end
 
   private
 
+  # Executes callbacks meant to be invoked after duplicated objects have been saved
+  #
+  # @param [Object|Array] _objects The source object(s)
+  # @param [Object|Array] duplicated The duplicated object(s)
+  # @return [Boolean] true if all callbacks are executed successfully
+  def after_save(_objects, duplicated)
+    populate_assessment_category_default_tabs(duplicated)
+  end
+
+  # Creates the mandatory tab for duplicated assessment categories if no tabs are already present.
+  # This cannot be done when duplicating the category since we do not know if a tab belonging to it will
+  # be duplicated afterward.
+  #
+  # @param [Object|Array] duplicated The duplicated object(s)
+  # @return [Boolean] true if all tabs are added successfully
+  def populate_assessment_category_default_tabs(duplicated)
+    duplicates_array = duplicated.respond_to?(:to_ary) ? duplicated : [duplicated]
+    tabless_categories = duplicates_array.select do |object|
+      object.class == Course::Assessment::Category && object.reload.tabs.empty?
+    end
+    tabless_categories.all? do |category|
+      default_tab = category.build_initial_tab
+      default_tab.creator = category.creator
+      default_tab.updater = category.updater
+      default_tab.created_at = category.created_at
+      default_tab.updated_at = category.updated_at
```

Comment:
shall we leave these as current user and current time?

Reply:
that works too, fixed with:
```
    User.with_stamper(duplicator.options[:current_user]) do
      tabless_categories.all? do |category|
        category.build_initial_tab
        category.save
      end
    end
```